compose/project.py

```diff
@@ -360,7 +389,7 @@ def matches_service_names(container):
                 self.service_names,
             )
 
-        return [c for c in containers if matches_service_names(c)]
+        return [c for c in containers if matches_service_names(c, orphan)]
```

Comment:
Since this is a nested function you actually already have access to `orphan` from the parent scope, and don't need to pass it in here.  You can drop the extra parameter to `matches_service_names()` as well.